frontend/src/components/modal.js

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Form } from 'reactstrap';
+import {  FormGroup, Label, Input } from 'reactstrap';
+import axios from 'axios'
+
+
+class ModalExample extends React.Component {
+  
+    state = {
+        modalState: false,
+        casc: false,
+        title: '',
+        warVersion: '', 
+        artifactID: '',
+        description: '',
+        dockertag:'',
+        dockerBase:'',
+    }
+
+    constructor(props) {
+        super(props);
+        this.toggle = this.toggle.bind(this);
+    }
+
+    componentDidMount() {
+        this.setState({modalState: this.props.modalState})
+        this.props.setClick(this.toggle);
+     }
+
+    toggle = () => {
+        this.setState({modalState: !this.state.modalState})
+    }
+
+    generateJSON = () => {
+        console.log("Generating JSON")
+        var packagerInfo = {}
+        // Create a bundle object
+        var bundle = new Object()
+        bundle["artifactId"] = this.state.artifactID
+        bundle["title"] = this.state.title
+        bundle["desc"] = this.state.description
+        packagerInfo["bundle"] = bundle
+        // Add war object
+        packagerInfo["war"] = {jenkinsVersion: this.state.warVersion}
+        packagerInfo["casc"] = true
+        // Add plugin object
+        packagerInfo["plugins"] = JSON.parse(localStorage.getItem("pluginsArray"))
+        // Add buildSettings object
+        var docker = {}
+        // docker["build"] = true
+        docker["tag"] = this.state.dockertag
+        docker["base"] = this.state.dockerBase
+        packagerInfo["buildSettings"] = docker
+        // Add default System Settings
+        var sysSettings = {}
+        sysSettings["setupWizard"] = "true"
+        sysSettings["slaveAgentPort"] = "5000"
+        sysSettings["slaveAgentPortEnforce"] = "true"
+        packagerInfo["sysSettings"] = sysSettings
+        localStorage.setItem("packageConfigJSON", JSON.stringify(packagerInfo))
+        this.submitConfiguration();
+    }
+
+    submitConfiguration = async () => {
+        console.log("Submitting configuration")
+        const apiURL = "http://localhost:8080/package/getPackageConfiguration";
```

Comment:
This should get pulled out into a variable eventually since the main way you're trying to run the tool is through the docker-compose.  I think that eventually you should pull this out to a config file.  Let's create a ticket for that work.

Reply:
Yeah adding a ticket :+1: